src/main/java/org/opentest4j/MultipleFailuresError.java

```diff
@@ -34,15 +36,15 @@
 
 	private static final long serialVersionUID = 1L;
 
-	private final List<AssertionError> failures = new ArrayList<AssertionError>();
+	private final List<Throwable> failures = new ArrayList<Throwable>();
 
 	private final String heading;
 
 	public MultipleFailuresError(String heading) {
 		this.heading = isBlank(heading) ? "Multiple Failures" : heading.trim();
 	}
 
-	public void addFailure(AssertionError failure) {
+	public void addFailure(Throwable failure) {
```

Comment:
This change was motivated by the needs of JUnit5 vintage engine to accumulate multiple failures.

The addFailure pre-existed so I did not think at all about changing it. And its mutability was actually convenient to gather the failures

In the context of my change we could change vintage TestRun, executionResults field from a map from TestDescriptor to TestExecutionResult to map to a list of results and create the Error with the constructor during getStoredResult.

I am not sure though where else this is used and/or it's intended use, so there might be other cases where it is required.